### ISSUE TITLE
`tctl rm` deletes two users at once

### ISSUE THREAD

**Issue by bpdohall** [gravitational/teleport#12328]
Given a Teleport instance having two users, `admin` and `admin@domain.com`.
Login  as `admin@domain.com`.
Execute `tctl users rm admin`.

Expected behavior:
Expect that only the `admin` user is deleted, not the user having an email address where the alias component matches the username provided to `tctl` 

Current behavior:
Both `admin` and `admin@domain.com` are deleted. The current session becomes invalid.

Bug details:
- Teleport version:  8.3.7
- Recreation steps
```
$ tsh --proxy=brendan.cloud.gravitational.io login
Enter password for Teleport user brendan.ohalloran@goteleport.com:
...snip...
$ tctl users ls
User                             Roles
-------------------------------- ---------------------
brendan.ohalloran                access,auditor,editor
brendan.ohalloran@goteleport.com access,auditor,editor

$ tctl users rm brendan.ohalloran
User "brendan.ohalloran" has been deleted
$ tctl users ls
[CLIENT]       Cannot connect to the auth server: failed direct dial to auth server: Get "https://teleport.cluster.local/v2/configuration/name": x509: certificate is valid for *.brendan.cloud.gravitational.io, brendan.cloud.gravitational.io, not 6272656e64616e2e636c6f75642e6772617669746174696f6e616c2e696f.teleport.cluster.local
	Get "https://teleport.cluster.local/v2/configuration/name": x509: certificate is valid for *.brendan.cloud.gravitational.io, brendan.cloud.gravitational.io, not 6272656e64616e2e636c6f75642e6772617669746174696f6e616c2e696f.teleport.cluster.local, failed dial to auth server through reverse tunnel: auth API: access denied [00]
	auth API: access denied [00].
Is the auth server running on "brendan.cloud.gravitational.io:443"?
```
- Debug logs
  - [tctl rm debug logs.txt](https://github.com/gravitational/teleport/files/8592361/tctl.rm.debug.logs.txt)

**Comment by r0mant** [comment]
@bpdohall This sounds like a duplicate of https://github.com/gravitational/teleport/issues/5188 which we have on the roadmap to fix. Thanks for the report!

**Comment by capnspacehook** [comment]
This was fixed with #12726.
